sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueListener.cs

```diff
@@ -322,7 +322,8 @@ internal async Task ProcessMessageAsync(QueueMessage message, TimeSpan visibilit
                 }
 
                 FunctionResult result = null;
-                using (ITaskSeriesTimer timer = CreateUpdateMessageVisibilityTimer(_queue, message, visibilityTimeout, _exceptionHandler))
+                Action<UpdateReceipt> onUpdateReceipt = updateReceipt => { message = message.Update(updateReceipt); };
```

Comment:
Can you help me with a bit of context around why do we need this?

Reply:
This has fixed RenewedQueueMessage_DeletesCorrectly test which isn't really related to base64 change.
In track 1 `CloudQueueMessage` was mutable and updating visibility timeout resulted in popreceipt being updated in message passed to SDK, so the code could keep original message and use it to delete message from queue if processing eventually finished after timeout and message hasn't been grabbed from the queue meanwhile.

In track 2 QueueMessage is read-only, but offers Update API that copies old message and updates with new receipt. So I had to propagate updated message to the caller to achieve same result.

Reply:
Ok, I think I understand what it's doing.